motionplan/motionPlanner.go

```diff
@@ -54,10 +54,57 @@ type PlanRequest struct {
 	Options            map[string]interface{}
 }
 
+// validatePlanRequest ensures PlanRequests are not malformed.
+func (req *PlanRequest) validatePlanRequest() error {
+	if req.Logger == nil {
```

Comment:
When would the logger be nil? I don't think we should create a new logger if the logger is nil. We should either return an error or panic as the logger should be considered a required field. I prefer the latter.

Reply:
We really ought not be panicking unless we absolutely cannot avoid it; an error message seems like it would be in the spirit of the Validate function.